uPortal-core/src/main/java/org/apereo/portal/portlet/om/PortletLifecycleState.java

```diff
@@ -15,21 +15,34 @@
 package org.apereo.portal.portlet.om;
 
 public enum PortletLifecycleState {
+
     CREATED(0),
     APPROVED(1),
     PUBLISHED(2),
     EXPIRED(3),
     MAINTENANCE(4);
 
-    public static final String MAINTENANCE_MODE_PARAMETER_NAME =
-            "PortletLifecycleState.inMaintenanceMode";
-
     private int order;
 
-    private PortletLifecycleState(int order) {
+    PortletLifecycleState(int order) {
         this.order = order;
     }
 
+    public static PortletLifecycleState forOrderValue(int orderValue) {
+        PortletLifecycleState rslt = null;
+        for (PortletLifecycleState state : PortletLifecycleState.values()) {
+            if (state.getOrder() == orderValue) {
+                rslt = state;
+                break;
+            }
+        }
+        if (rslt == null) {
```

Comment:
if `state` is `null` won't an exception be thrown on line 34 `state.getOrder()`?